src/hooks/use-drag-and-drop-file.ts

```diff
@@ -36,10 +36,26 @@ export function useDragAndDropFile< T extends HTMLElement >( {
 			if ( event.dataTransfer.files.length === 1 ) {
 				onFileDrop( event.dataTransfer.files[ 0 ] );
 			}
+			setIsDraggingOver( false );
+		};
+		const handleDragEnd = ( event: DragEvent ) => {
+			event.preventDefault();
+			setIsDraggingOver( false );
+		};
+
+		const node = dropRef.current;
+
+		node.addEventListener( 'dragover', handleDragOver );
+		node.addEventListener( 'dragleave', handleDragLeave );
+		node.addEventListener( 'drop', handleDrop );
+		node.addEventListener( 'dragend', handleDragEnd );
+
+		return () => {
+			node.removeEventListener( 'dragover', handleDragOver );
+			node.removeEventListener( 'dragleave', handleDragLeave );
+			node.removeEventListener( 'drop', handleDrop );
+			node.removeEventListener( 'dragend', handleDragEnd );
 		};
-		dropRef.current.addEventListener( 'dragover', handleDragOver );
-		dropRef.current.addEventListener( 'dragleave', handleDragLeave );
-		dropRef.current.addEventListener( 'drop', handleDrop );
 
 		function cleanup() {
 			if ( ! dropRef.current ) {
```

Comment:
If we return the remove listeners directly, then we may want to remove the cleanup function and the second return.

Reply:
I kept this original logic as after some testing, it worked well and removed further modifications in https://github.com/Automattic/studio/pull/378/commits/cdcac1b629a7b1b877725714021416e0368fb347